website/docs/language/expressions/custom-conditions.mdx

```diff
@@ -13,9 +13,14 @@ You can create conditions that produce custom error messages for several types o
 This page explains the following:
   - Creating [validation conditions](#input-variable-validation) for input variables (Terraform v0.13.0 and later)
   - Creating [preconditions and postconditions](#preconditions-and-postconditions) for resources, data sources, and outputs (Terraform v1.2.0 and later)
+  - Creating [assertions](#check-assertions) within check blocks (Terraform v1.5.0 and later)
   - Writing effective [condition expressions](#condition-expressions) and [error messages](#error-messages)
   - When Terraform [evaluates custom conditions](#conditions-checked-only-during-apply) during the plan and apply cycle
 
+The various custom conditions are suited to different situations. Input variable validation and output postconditions should be used to validate the inputs and outputs of your configuration meet any requirements or conditions. Resource preconditions and postconditions should be used to validate your configuration is produced expected and predictable results. Check blocks and assertions should be used to evaluate your infrastructure as a whole.
```

Comment:
Now that I look at it, it feels like this  paragraph should be it's own section? How do you feel about something like the following? I also reformatted this paragraph as a list because it felt easier to parse, and rearranged the order of options to feature checks first because it one of the broadest use cases:

```suggestion
## Selecting a Custom Condition for Your Use Case

Terraform's different custom conditions are best suited to various situations. Use the following broad guidelines to select the best custom condition for your use case:
1. [Check blocks with assertions](#check-assertions) validate your infrastructure as a whole.
1. [Validation conditions](#input-variable-validation) or [output postconditions](#preconditions-and-postconditions) can ensure your configuration's inputs and outputs meet specific requirements.
1. Resource [preconditions and postconditions](#preconditions-and-postconditions) can validate that Terraform produces your configuration with predictable results.
```